src/components/widgets/sales-cycle/sales-cycle.directive.coffee

```diff
@@ -22,25 +22,22 @@ module.controller('WidgetSalesCycleCtrl', ($scope, $q, ChartFormatterSvc, $filte
   # Widget specific methods
   # --------------------------------------
   w.initContext = ->
-    if $scope.isDataFound = angular.isDefined(w.content) && !_.isEmpty(w.content.status_average_durations)
+    dhb = ImpacDashboardsSvc.getCurrentDashboard()
+    status_selection = w.metadata.status_selection || dhb.metadata.status_selection || { values: [] }
 
-      # Takes metadata from dashboard or from widget according to 'reach' parameter
-      status_selection = if w.metadata.status_selection && w.metadata.status_selection.reach == 'dashboard' then ImpacDashboardsSvc.getCurrentDashboard().status_selection else w.metadata.status_selection
-      status_selection = status_selection || {values:[]}
+    if $scope.isDataFound = angular.isDefined(w.content) && !_.isEmpty(w.content.status_average_durations)
+      # Remove statuses absent from statuses list returned by the widget engine
+      _.remove status_selection.values, (status) ->
+        status not in _.keys w.content.status_average_durations
```

Comment:
This avoids having an empty widget if the status_selection has only elements that won't be recognized by the widget (case where sales cycle and leads funnel have different statuses)